src/core/sagas/recommendations.ts

```diff
@@ -0,0 +1,54 @@
+import 'isomorphic-fetch';
+import * as effects from 'redux-saga/effects';
+import FluxCapacitor from '../../flux-capacitor';
+import Actions from '../actions';
+import Adapter from '../adapters/search';
+import Configuration from '../configuration';
+import Selectors from '../selectors';
+import Store from '../store';
+
+declare function fetch();
+
+export namespace Tasks {
+  // tslint:disable-next-line max-line-length
+  export function* fetchProducts(flux: FluxCapacitor, action: Actions.FetchRecommendationsProducts) {
+    try {
+      const state = yield effects.select();
+      const config = flux.config.recommendations;
+      const mode = Configuration.RECOMMENDATION_MODES[config.mode || 'popular'];
+      // tslint:disable-next-line max-line-length
+      const recommendationsUrl = `https://${flux.config.customerId}.groupbycloud.com/wisdom/v2/public/recommendations/products/_get${mode}`;
+      const recommendationsResponse = yield effects.call(fetch, recommendationsUrl, {
+        method: 'POST',
+        body: JSON.stringify({
+          size: config.productCount,
+          type: 'viewProduct',
+          target: 'productId'
```

Comment:
target is not always productId it's sometimes SKU -> should add configuration for this option so clients can change it depending on their data